src/content-proxy.js

```diff
@@ -47,20 +63,27 @@ async function contentProxy(opts) {
   url.searchParams.append('rid', options.requestId);
 
   log.info(`Fetching content from: ${url.href}`);
-  const response = await fetch(url.href, getFetchOptions(options));
-  const body = await response.text();
-  if (response.ok) {
+  const resp = await fetch(url.href, getFetchOptions(options));
+  const body = await resp.buffer();
+  if (resp.ok) {
+    const headers = {};
+    PASSTHROUGH_HEADERS.forEach((name) => {
+      const value = resp.headers.get(name);
+      if (value) {
+        headers[name] = value;
+      }
+    });
     return new Response(body, {
       status: 200,
-      headers: response.headers,
+      headers,
     });
   }
-  log[utils.logLevelForStatusCode(response.status)](`Unable to fetch ${url.href} (${response.status}) from content-proxy: ${body}`);
+  log[utils.logLevelForStatusCode(resp.status)](`Unable to fetch ${url.href} (${resp.status}) from content-proxy: ${body}`);
   return new Response(body, {
-    status: utils.propagateStatusCode(response.status),
+    status: utils.propagateStatusCode(resp.status),
     headers: {
-      'x-error': response.headers.get('x-error'),
-      vary: response.headers.get('vary'),
+      'x-error': resp.headers.get('x-error'),
+      vary: resp.headers.get('vary'),
       'cache-control': 'max-age=60',
```

Comment:
why setting the cache control ?

Reply:
I took both from code in helix-content-proxy where I assumed that it needs to set those headers in case of an error. I can remove them if not needed.

Reply:
@trieloff WDYT?

Reply:
but this is the reply you would send back to who ever invokes the content-bus. for example the sidekick.
I don't think you need any vary header. rather always set cache headers to never caceh.